test/unit/rules.spec.ts

```diff
@@ -0,0 +1,51 @@
+import {expect} from 'chai';
+import {
+  ResourceGraphRule,
+  ResourceGraphExecutor,
+  RuleType,
+} from '../../src/rules';
+import {ResourceGraphModels} from '@azure/arm-resourcegraph';
+import {HttpHeadersLike, WebResourceLike} from '@azure/ms-rest-js';
+
+describe('Resource Graph Rule', () => {
+  const mockResourcesResponse: ResourceGraphModels.ResourcesResponse = {
+    totalRecords: 1,
+    count: 1,
+    resultTruncated: 'false',
+    data: {columns: [{name: 'id', type: 'string'}], rows: [['mockResourceId']]},
+    _response: {
+      request: {} as WebResourceLike,
+      status: 200,
+      headers: {} as HttpHeadersLike,
+      bodyAsText: '',
+      parsedBody: {} as ResourceGraphModels.QueryResponse,
+    },
+  };
+  const mockRule: ResourceGraphRule = {
+    name: 'test-rule',
+    query: 'mock query',
+    description: 'Intentional bad query',
+    type: RuleType.ResourceGraph,
+  };
+  it('can produce a scan result', async () => {
+    const scanResult = ResourceGraphExecutor.toScanResult(
+      mockResourcesResponse,
+      mockRule
+    );
+    expect(scanResult).to.deep.equal({
+      ruleName: mockRule.name,
+      description: mockRule.description,
+      total: 1,
+      resourceIds: ['mockResourceId'],
+    });
+  });
+  it("should throw an errow if the 'id' column is not returned from Resource Graph", async () => {
+    mockResourcesResponse.data.columns = [];
```

Comment:
We're mutating the global test data here. If tests run in parallel, and/or if we add more tests, then results become unpredictable. 2 quick ideas

1: create the specific test data you need inside each test case
2: change from a global const to a function that returns a new object every time, and then you can do whatever you want to modify that new object w/o affecting other tests (assuming that function isn't also using globals for its child properties, etc)